### PR TITLE
Fix Publish CI

### PR DESCRIPTION
**Pull request by sudo-shashank**
**Summary of changes**
Changes introduced in this pull request:
- Publish CI fails to publish all crates from a single runner giving `No Space left of device` [error](https://github.com/ChainSafe/fil-actor-states/actions/runs/9771997028/attempts/1).
This PR separates the publishing actors crate from the actor interface and enables parallel publishing of crates while maintaining their inter-dependencies.

**Reference issue to close (if applicable)**

**Other information and links**

### CODE REVIEW

**Review by LesnyRumcajs — CHANGES_REQUESTED** [review]
As discussed on Slack, let's resolve the issue differently.

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -5,20 +5,149 @@ on:
   push:
     tags:
       - v*
+
 env:
   RUSTC_WRAPPER: "sccache"
```
Comment (hanabi1224) [inline .github/workflows/publish.yml:10]: let's add `SCCACHE_GHA_ENABLED: "true"`

Comment (sudo-shashank) [reply .github/workflows/publish.yml:10]: fixed

**Review by hanabi1224 — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -5,20 +5,163 @@ on:
...
+        ]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Setup sccache
+        uses: mozilla-actions/sccache-action@v0.0.5
+        timeout-minutes: 5
+        continue-on-error: true
+        env:
+          SCCACHE_GHA_ENABLED: "true"
```
Comment (hanabi1224) [inline .github/workflows/publish.yml:123]: `SCCACHE_GHA_ENABLED` can be under the global `env:` no?

**Review by LesnyRumcajs — COMMENTED** [review]
Could we use [composite action](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action) to reduce code duplication?

File: .github/workflows/publish.yml (hunk 2 above)
Comment (sudo-shashank) [reply .github/workflows/publish.yml]: yes

**Comment by sudo-shashank** [comment]
> Could we use [composite action](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action) to reduce code duplication?

@LesnyRumcajs yes updated

File: .github/composite-actions/publish-crate/action.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      timeout-minutes: 5
+      continue-on-error: true
+    - name: Publish crate
+      run: cargo publish --package ${{ inputs.crate }} --token ${{ env.CRATES_IO_TOKEN }}
+    - name: Clear artifacts
+      run: cargo clean
+inputs:
+  crate:
+    description: "The crate to publish"
+    required: true
```
Comment (LesnyRumcajs) [inline .github/composite-actions/publish-crate/action.yml:19]: Let's have the `inputs` before `run` so that it's more visible.

**Review by LesnyRumcajs — APPROVED** [review]

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**